### PR TITLE
add tls and basic auth support

### PR DESCRIPTION
**Pull request by lmtyler**

### CODE REVIEW

File: examples/stan-bench/main.go [hunk 1]
```diff
@@ -49,6 +55,65 @@ var (
...
+
+	return nil
+}
+
+func AddCertFromFile(certPool *x509.CertPool, file string) error {
+	if len(file) == 0 {
+		return fmt.Errorf("invalid file name")
+	}
+
+	if strings.HasSuffix(file, ".pem") == false {
```
Comment (kozlovic) [inline examples/stan-bench/main.go:100]: You need to change this to `if !strings.HasSuffix(..`. See https://travis-ci.org/github/nats-io/stan.go/builds/703910822#L265

File: examples/stan-bench/main.go [hunk 2]
```diff
@@ -83,6 +158,22 @@ func main() {
 		opts = append(opts, nats.UserCredentials(*userCreds))
 	}
 
+	// Use BasicAuth
+	if len(user) > 0 && len(pswd) > 0 {
+		opts = append(opts, nats.UserInfo(user, pswd))
+	}
+
+	if strings.Contains(*urls, "tls://") == true {
```
Comment (kozlovic) [inline examples/stan-bench/main.go:166]: Omit `== true`. See https://travis-ci.org/github/nats-io/stan.go/builds/703910822#L266.

File: examples/stan-bench/main.go [hunk 3]
```diff
@@ -66,6 +131,16 @@ func main() {
 	var csvFile = flag.String("csv", "", "Save bench data to csv file")
 	var queue = flag.String("qgroup", "", "Queue group name")
 	var userCreds = flag.String("creds", "", "Credentials File")
+	var certDir string
```
Comment (kozlovic) [inline examples/stan-bench/main.go:134]: Why not stay with the existing code style and have declaration like:
```
var certDir = flag.String("cd", ...
```
and then use `*certDir` to get the string?
(same for all others you have added)

File: examples/stan-bench/main.go [hunk 4]
```diff
@@ -49,6 +55,65 @@ var (
 	qSubsLeft  int32
 )
 
+func CreateTlsConfig(dirName string) (*tls.Config, error) {
```
Comment (kozlovic) [inline examples/stan-bench/main.go:58]: I don't think you need that one: `nats.RootCAs(file ...string)` does the same for given files. So if anything you may just need to collect the cert files?

**Comment by lmtyler** [comment]
Missed nats.RootCAs
made other requested changes

**Pull request merged**